src/backend/distributed/utils/datautils.c

```diff
@@ -0,0 +1,32 @@
+/*-------------------------------------------------------------------------
+ *
+ * datautils.c
+ *
+ * This file contains functions to perform useful operations on data types.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "c.h"
+
+#include "distributed/datautils.h"
+
+
+/*
+ * Allocates a new uint64 on the heap and copies the input to that space,
+ * returning a pointer to allocated memory. Useful for lists with 64-bit
+ * integers.
+ */
+uint64 *
+AllocateUint64(uint64 value)
+{
+	uint64 *allocatedValue = (uint64 *) palloc0(sizeof(uint64));
```

Comment:
> It does

I was searching the symbol index on my PostgreSQL `REL9_5_STABLE` checkout, which doesn't include this function (i.e. it's not yet in a released PostgreSQL). So we'd have to polyfill it anyhow.

> You can just use a non-int list. Datums are guaranteed to be storable in
a size_t wide variable.

Interesting. I was missing this piece of knowledge. Now I see how this could work, thanks!